prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirToHl7Converter.kt

```diff
@@ -166,6 +166,11 @@ class FhirToHl7Converter(
                 }
             }
         }
+
+        // when valueSet is available, use the matching value else just pass the value as is
+        // does a lowerCase comparison
+        if (element.valueSet.isNotEmpty()) retVal = element.valueSet.getOrDefault(retVal.lowercase(), retVal)
```

Comment:
Will this create problems if the value set is written in uppercase and the values come in as lowecase?